helix-core/src/main/java/org/apache/helix/manager/zk/ZkSessionAwareStateListener.java

```diff
@@ -0,0 +1,27 @@
+package org.apache.helix.manager.zk;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.I0Itec.zkclient.IZkStateListener;
+
+
+public interface ZkSessionAwareStateListener extends IZkStateListener {
```

Comment:
Will add comments for this public interface.

Reply:
Why do we need this?

Reply:
If we have to introduce this new method, why not add one method into existing IZkStateListener (with default implementation just calling the old one in the interface definition)?

Reply:
We need a new api: handleNewSession(sessionId). 

handleNewSession() is defined in IZkStateListener, which is an interface defined in I0Itec.zkclient. We can not just change handleNewSession() with handleNewSession(sessionId) in I0Itec.zkclient. Considering backward compatibility, we need to extend interface IZkStateListener to add a new api: handleNewSession(sessionId). The benefit is subscriptions of  IZkStateListener don't have to change.

So with this new extended interface, we can both support backward compatibility and have no changes to stateListener subscriptions.

Reply:
I think we are getting rid of I0Itec when we are moving all ZK code to the new zkclient module. If that is the case, should we wait for that to make the interface clear (and less confusing)?

Reply:
Regarding this, we may discuss further. I think that would be a good idea to wait for that, if we can wait for that in reasonable time.